locator/token_metadata.cc

```diff
@@ -1035,8 +1037,19 @@ token_metadata::clone_after_all_left() const noexcept {
     co_return token_metadata(co_await _impl->clone_after_all_left());
 }
 
+void token_metadata::clear_and_dispose_impl() noexcept {
+    if (!_shared_token_metadata) {
+        return;
```

Comment:
Shouldn't we _impl->clear_gently() in that case in token_metadata::clear_gently()?